Add public convenience API for generating a transport event with product data (#108)
google/GoogleDataTransport: CHANGELOG.md

# Unreleased

> - Added new APIs to support attaching a product ID data to an event.

Reviewer: Consider "Added new APIs for attaching a product ID data to an event.

Author: ```suggestion
- Added new APIs for attaching product ID data to an event.
```

Author: Thanks, that's better.
